mtec/mtec.h

```diff
@@ -0,0 +1,128 @@
+* Alternatively, this project may be redistributed and/or modified under the
+* terms of the GNU Lesser General Public License as published by the Free
+* Software Foundation; version 3. This project is distributed in the hope that
+* it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this project. If not, see <https://www.gnu.org/licenses/>.
+*
+* For any further details and any questions please contact us under
+* contact@nymea.io or see our FAQ/Licensing Information on
+* https://nymea.io/license/faq
+*
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#ifndef MTEC_H
+#define MTEC_H
+
+#include <QDateTime>
+#include <QObject>
+
+#include "../modbus/modbustcpmaster.h"
+
+#include "mtecinfo.h"
+
+class MTec : public QObject
+{
+    Q_OBJECT
+public:
+    /** Constructor */
+    explicit MTec(const QHostAddress &address, QObject *parent = nullptr);
+
+    /** Destructor */
+    ~MTec();
+
+    QHostAddress getHostAddress() const {return m_hostAddress;};
+
+private:
+    /**
+     * The first response of the M-Tec heatpump may be delayed
+     * by 10 s. The plugin will wait for the defined time in milliseconds
+     * for a response before sending an additional request or setting
+     * an error code.
+     */
+    static const int  FirstConnectionTimeout = 15000;
+
+    /** Modbus Unit ID (undocumented, guessing 1 for now) */
+    static const quint16 ModbusUnitID = 1;
+
+    /** The following modbus addresses can be read: */
+    enum RegisterList {
+        /**
+         * APPL.CtrlAppl.sParam.heatpump[0].ElectricEnergyMeter.values.power
+         * Actual power consumtion [W]
+         */
+        ActualPowerConsumption              = 707,
+
+        /**
+         * APPL.CtrlAppl.sIOModule.Virt[0].param.sensor[0]
+         * Acutal excess energy given from Smart home System [W]
+         */
+        ActualExcessEnergySmartHome         = 1000,
+
+        /**
+         * APPL.CtrlAppl.sParam.photovoltaics.ElectricEnergyMeter.values.power
+         * Acutal excess energy given from Electricity Meter [W]
+         */
+        ActualExcessEnergyElectricityMeter  = 1002,
+
+        /**
+         * APPL.CtrlAppl.sParam.extHeatSource[0].param.externalSetvalueScaled
+         * Control of the heat source by an external control [100%]
+         */
+        ExternalSetValueScaling             = 1600,
+
+        /**
+         * APPL.CtrlAppl.sParam.extHeatSource[0].values.requestExtHeatsource
+         * 0…no request, external heat source must be turned off.
+         * 1…external heat source is released and can be switched on.
+         * 2…external heat source is required and must be turned on.
+         */
+        RequestExternalHeatSource           = 1601
+    };
+
```

Comment:
Add `Q_ENUM(RegisterList)`, so you will have it available in the meta object and you can do stuff like
`qCDebug(dcMTec()) << register;`

Reply:
Done